devito/tools/dtypes_lowering.py

```diff
@@ -145,6 +145,9 @@ def dtype_to_ctype(dtype):
         # Bypass np.ctypeslib's normalization rules such as
         # `np.ctypeslib.as_ctypes_type(ctypes.c_void_p) -> ctypes.c_ulong`
         return dtype
+    elif dtype == np.float16:
```

Comment:
I think this should be fine since it will only catch the types before they're mapped to ours; numpy docs suggest using equality for checking dtypes if you mean the `==` is dangerous as opposed to `is`. But yes will move to allocator